packages/create-twilio-function/src/create-twilio-function/create-gitignore.js

```diff
@@ -1,20 +1,25 @@
-const fs = require('fs');
+const fs = require('fs').promises;
+const { createWriteStream } = require('fs');
 const path = require('path');
 const { promisify } = require('util');
 
 const writeGitignore = promisify(require('gitignore').writeFile);
 
-const open = promisify(fs.open);
+const ADDITIONAL_CONTENT = `
+# Twilio Serverless
+.twiliodeployinfo
+`;
 
-function createGitignore(dirPath) {
+async function createGitignore(dirPath) {
   const fullPath = path.join(dirPath, '.gitignore');
-  return open(fullPath, 'wx').then((fd) => {
-    const stream = fs.createWriteStream(null, { fd });
-    return writeGitignore({
-      type: 'Node',
-      file: stream,
-    });
+  const fd = await fs.open(fullPath, 'wx');
+  const stream = createWriteStream(fullPath, { fd });
+  await writeGitignore({
+    type: 'Node',
+    file: stream,
   });
+  await fd.close();
+  await fs.appendFile(fullPath, ADDITIONAL_CONTENT, 'utf8');
```

Comment:
Rather than close the file descriptor and open it again, you could do:

```suggestion
await fd.appendFile(ADDITIONAL_CONTENT, 'utf8');
await fd.close();
```

Reply:
That unfortunately didn't work. I've tried that for a while and gave up because the other thing worked.

Reply:
Hmm, you're right. That's because the `gitignore` package closes the stream itself. And perhaps the open file descriptor. I'm having a go at rewriting this because the confusion between promises and streams (which don't support promises in the version of Node we are using are making this harder to deal with. PR incoming.